sdk/core/azure-core/src/test/java/com/azure/core/http/policy/HttpLogOptionsTest.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.http.policy;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class HttpLogOptionsTest {
+
+    @Test
+    public void testMaxApplicationId() {
+        IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
+            () -> new HttpLogOptions().setApplicationId("AppId-0123456789012345678912345"));
+        assertEquals(thrown.getMessage(), "'applicationId' length cannot be greater than 24");
```

Comment:
Asserting the actual message isn't a good idea because this could be localised or message changed which would break the test.